skimage/filters/ridges.py

```diff
@@ -0,0 +1,643 @@
+"""
+Ridge filters.
+
+Ridge filters can be used to detect continuous edges, such as vessels,
+neurites, wrinkles, rivers, and other tube-like structures. The present
+class of ridge filters relies on the eigenvalues of the Hessian matrix of
+image intensities to detect tube-like structures where the intensity changes
+perpendicular but not along the structure.
+"""
+
+from itertools import combinations_with_replacement
+
+from warnings import warn
+
+import numpy as np
+
+from ..util import img_as_float, invert
+from ..feature import hessian_matrix, hessian_matrix_eigvals
+from .._shared.utils import assert_nD
+
+
+def divide_nonzero(array1, array2, cval=1e-10):
```

Comment:
I'm more strongly for keeping this auxiliary function since it would involve quite lengthy commands otherwise (e.g. np.where(array2==0, array1/array2, array1*1e10)) . Or is this still preferred, or do you have other suggestions?